internal/tiltfile/starkit/path.go

```diff
@@ -19,15 +19,11 @@ func AbsWorkingDir(t *starlark.Thread) string {
 	return filepath.Dir(CurrentExecPath(t))
 }
 
-// Path to the file at the bottom of the call stack.
+// Path to the file that's currently executing
 func CurrentExecPath(t *starlark.Thread) string {
-	depth := t.CallStackDepth()
-	for i := 0; i < depth; i++ {
-		filename := t.CallFrame(i).Pos.Filename()
-		if filename == "<builtin>" {
-			continue
-		}
-		return filename
+	ret := t.Local(execingTiltfileKey)
+	if ret == nil {
```

Comment:
i'm a little bit worried that we have deferred functions (like workloadToResourceFn) that aren't called from an active starlark thread, but maybe we should just change that API so everything has a real thread

Reply:
Good call. This breaks with workload_to_resource_function, but that also breaks in master, so not going to block this PR on it. I've filed https://app.clubhouse.io/windmill/story/5863/workload-to-resource-function-does-not-use-a-proper-starkit-thread-which-can-cause-internal-errors.